astropy/units/quantity.py

```diff
@@ -314,6 +315,96 @@ class Quantity(np.ndarray):
+        ----------
+        unit_shape_dtype : :class:`~astropy.units.UnitBase`, str, `~astropy.units.PhysicalType`, or tuple
+            Unit specification, can be the physical type (ie str or class).
+            If tuple, then the first element is the unit specification
+            and all other elements are for `numpy.ndarray` type annotations.
+            Whether they are included depends on the python and :mod:`numpy`
+            versions.
+
+        Returns
+        -------
+        `typing.Annotated`, `typing_extensions.Annotated`, `astropy.units.Unit`, or `astropy.units.PhysicalType`
+            Return type in this preference order:
+            `typing.Annotated` if python v3.9+; `typing_extensions.Annotated`
+            if :mod:`typing_extensions` is installed, a `astropy.units.Unit`
+            or `astropy.units.PhysicalType` else-wise, depending on the input.
+
+        Raises
+        ------
+        TypeError
+            If the unit/physical_type annotation is not Unit-like or
+            PhysicalType-like.
+
+        Examples
+        --------
+        Create a unit-aware Quantity type annotation
+
+            >>> Quantity[Unit("s")]
+            Annotated[Quantity, Unit("s")]
+
+        See Also
+        --------
+        `~astropy.units.quantity_input`
+            Use annotations for unit checks on function arguments and results.
+
+        Notes
+        -----
+        With Python 3.9+ or :mod:`typing_extensions`, |Quantity| types are also
+        static-type compatible.
+        """
+        # LOCAL
+        from ._typing import HAS_ANNOTATED, Annotated
+
+        # process whether [unit] or [unit, shape, ptype]
+        if isinstance(unit_shape_dtype, tuple):  # unit, shape, dtype
+            target = unit_shape_dtype[0]
+            shape_dtype = unit_shape_dtype[1:]
+        else:  # just unit
+            target = unit_shape_dtype
+            shape_dtype = ()
+
+        # Allowed unit/physical types. Errors if neither.
+        try:
+            unit = Unit(target)
+        except (TypeError, ValueError):
+            from astropy.units.physical import get_physical_type
+
+            try:
+                unit = get_physical_type(target)
+            except (TypeError, ValueError, KeyError):  # KeyError for Enum
+                raise TypeError("unit annotation is not a Unit or PhysicalType") from None
+
+        # Allow to sort of work for python 3.8- / no typing_extensions
+        # instead of bailing out, return the unit for `quantity_input`
+        if not HAS_ANNOTATED:
+            warnings.warn("Quantity annotations are valid static type annotations only"
+                          " if Python is v3.9+ or `typing_extensions` is installed.")
+            return unit
+
+        # Quantity does not (yet) properly extend the NumPy generics types,
+        # introduce in numpy v1.22+, instead just including the unit info as
+        # metadata using Annotated.
+        if minversion('numpy', '1.22'):
```

Comment:
Instead of importing `minversion`, you can
```
from astropy.utils.compat import NUMPY_LT_1_22
```
and then here use `if not NUMPY_LT_1_22`

That will help future grep'ing...

Reply:
Done!